### PR TITLE
Propose new stuff for ifdefs

### PR DESCRIPTION
**Pull request by ethteck**
We can discuss here or in Discord, but I realized I don't love the way the current "rules" for these are written and wanted to give my own opinion as to how they should look

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,11 +95,18 @@ Once you've matched a function, run the following:
...
+## Trouble matching a function
 
-If you make significant progress on a function but are unable to 100% match it, feel free to wrap it in a NON_EQUIVALENT check:
+Sometimes when working on a function, for various reasons, you'll be unable to match it. The following two sections describe what labels we give these functions and when. Any time you wrap a function in a label like this, please leave a comment above the function with a brief description of the issue(s).
+
+### NON_EQUIVALENT
+
+If you are working on a function but can't figure out how to tackle certain parts, or you're sure that it's not equivalent, you can wrap it in NON_EQUIVALENT.
+
+Please make a best-effort attempt to any function you put in a PR, though. Half-finished functions aren't necessarily much use to the next decomper.
```
Comment (nanaian) [inline CONTRIBUTING.md:106]: Probably missing the word "finish" here

Comment (ethteck) [reply CONTRIBUTING.md:106]: no, to any. what part of that don't you understand

Comment (ethteck) [reply CONTRIBUTING.md:106]: lol yeah good catch

Comment (nanaian) [reply CONTRIBUTING.md]: lol

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -109,6 +116,19 @@ INCLUDE_ASM(void, "file", func, void);
 #endif
 ```
 
-## NON_MATCHING
+### NON_MATCHING
+
+If you are unable to 100% match a function but think it's equivalent in behavior, you can wrap the C code in NON_MATCHING.
```
Comment (nanaian) [inline CONTRIBUTING.md:121]: This should maybe be "certain it's equivalent"?

**Review by nanaian — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (ethteck) [reply CONTRIBUTING.md:121]: I'll say "fairly certain" just cause I don't wanna scare people off too much. but yeah, stronger than "think" is good

**Review by Brotenko — APPROVED** [review]

**Review by JoshDuMan — APPROVED** [review]

**Pull request merged**